### PR TITLE
Generate README reading version from package.json

### PR DESCRIPTION
**Pull request by manueltanzi-okta**
Generate README reading version from package.json

Add grunt task to generate README.md file taking the version from package.json

Resolves : OKTA-135026

@hor-kanchan-okta @anishashetty-okta  @upteam-okta

### CODE REVIEW

**Comment by manueltanzi-okta** [comment]
RelEng needs to update promote script in order to make this work

File: scripts/publish.sh [hunk 1]
```diff
@@ -15,7 +15,10 @@ else
   TARGET_BRANCH=$BRANCH
 fi
 
-if ! ci-update-package --branch ${TARGET_BRANCH}; then
+if ci-update-package --branch ${TARGET_BRANCH};
+then
+  npm run generate-readme;
```
Comment (anishashetty-okta) [inline scripts/publish.sh]: Do we want to do this everytime we publish?

Comment (manueltanzi-okta) [reply scripts/publish.sh]: we update package.json so we can update also the README.md , but we should push this to github only on promotion

File: Gruntfile.js [hunk 2]
```diff
@@ -555,6 +555,13 @@ module.exports = function (grunt) {
     ));
   });
 
+  grunt.task.registerTask('generate-readme', 'Generate Readme.md file using package.json', function () {
+    var readmeTpl = grunt.file.read('./README_TPL.md');
+    var version = grunt.file.readJSON('package.json').version;
```
Comment (anishashetty-okta) [inline Gruntfile.js]: probably add a log line about the version being updated:
Something like:
`grunt.log.writeln('Updating version in README.md to ' + version);`

**Comment by rchild-okta** [comment]
Mentioned in IM, but let's hold off on this until we update the CDN process - currently the CDN version won't be available at time of promote because it ships out with the production schedule.

**Comment by jmelberg-okta** [comment]
Any reason we cannot do this now?

**Comment by haishengwu-okta** [comment]
@jmelberg-okta i think we could. I'm wondering if we could have simplify a little bit. e.g instead of generating readme from a template, can we change version in place in current README?

**Comment by aarongranick-okta** [comment]
Closing this PR as we can now update the README using a buildtool: https://github.com/okta/okta-signin-widget/pull/1697

**Pull request closed without merging**